sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java

```diff
@@ -149,8 +149,9 @@ public ReactorSession(AmqpConnection amqpConnection, Session session, SessionHan
 
         shutdownSignals = amqpConnection.getShutdownSignals();
         subscriptions.add(this.endpointStates.subscribe());
-        subscriptions
-            .add(shutdownSignals.flatMap(signal -> closeAsync("Shutdown signal received", null, false)).subscribe());
+        subscriptions.add(shutdownSignals
+            .flatMap(signal -> closeAsync("Shutdown signal received (" + signal.toString() + ")", null, false))
```

Comment:
toString() produces the shutdown message, logging it comes handy to understand what event in connection triggered session closure